### ISSUE TITLE
MKAP 2023 season: ideas / suggestions

### ISSUE THREAD

**Issue by luizaogs** [arch-kiosk/arch-kiosk-office#2309]
Here's a list of small things people have noticed this season that may be useful to consider going forward. Of course, if you guys disagree then ignore them! I'm opening this before I leave but if Didi runs into anything else in the week while I'm gone she will add to this ticket.

- for small finds periods: can we add a check box or something to indicate certain vs. uncertain dating? (I think I had already suggested this in regards to Edfu small finds processing)
- also for small finds periods, is there a way to make it possible to add more than one period? I know we can add a range to the configuration and pick that, but in some cases it is probably easier to just be able to pick more than one (instead of adding a bunch of different ranges)
- do you guys think it makes sense to have a way to copy closing elevations into the opening elevations of a new locus? those are often the same
- for relations, "is filled by" was missing from the dropdown menu. I managed to add it to the MKAP configuration, but since it's the only inverse relationship missing from the value list you probably need to add it to other projects. *** has been mostly taken care of in #2657 

Additions from 18 X 2023: 
- I think I also already mentioned this regarding Edfu, but what about a field for comparanda in cm details? we have a couple of bowls with comparanda from Lahun here, for instance, and we want to add that to the record. could add it to description, but would a separate thing not be more easily searchable? thinking of how it might be used in Q&V even
- Didi and I keep having to write down to-do lists on paper for every next day, in order to keep track of what we need to do. have you guys ever thought of adding that to the database? something simple but that could be updated every day and in which things could be checked off. I was thinking it might make sense on the unit level. *** ticket already exists: #2414

**Comment by lbestock** [comment]
Yay, notes from the field! First thoughts:

I have just checked on the "is filled by" on my own last downloaded copy of MKAP. Indeed it wasn't in the valuelist, though the inverse relationship and chronological relationship were configured. When I added it, however, it showed up as it should and behaved as it should. Can you check that again? Did it disappear immediately or only after synchronization? It is a list that automatically sorts alphabetically so if you put it at the end it won't show up at the end. If something added to a valuelist in configuration does not show up that is a way bigger problem than a locus relationship type missing. (@urapadmin it's also missing in KAMBE so wants updating in whatever you are using to prepare new Kiosks.)

I see problems with copying the elevation, though I also see why it is annoying to put the same value in twice when you have already defined the locus relationship between two loci. The problem is that you need more information. Locus 1 can be above three different loci, and which of its closing elevations are the opening elevations for which of those three loci? So at the very least you would need to have the choice on a locus by locus basis if you used closings for openings. I think what would help for this discussion is _how are you actually using elevations in analysis_? Whether that is in the field  interpreting something on the basis of elevations, or back from the field. I have never quite shaken my suspicion that elevations are something archaeologists mostly take by rote because we think we are supposed to, and I would think better about how to record them if I knew better how people actually use them. I do expect they matter more in a tell. But I want to hear more about their use and whether or not that is a use best supported by Kiosk or by the GIS.

**Comment by luizaogs** [comment]
For elevations: sorry, I should have been more specific. I didn't mean that a new locus would automatically get another's closing elevations. I think it would have to be on a locus by locus basis for sure, and then you would need to be able to pick which locus' elevations you want (which might be more complicated to implement than it's actually worth doing, of course). I end up not having time to add both closings and openings while trying to also take pictures and give instructions to workmen because doing so would mean flipping between locus forms and memorizing numbers that mean nothing to me, which means that I have a bunch of empty closings in the database (which I can go back and fill in later, but which I inevitably forget to do). I use elevations in the trench to figure out the profile. So for me it is mostly to write reports and interpret stratigraphy while in the field.

**Comment by luizaogs** [comment]
On the "is filled by" issue: guys, I'm so tired. That bullet point was written before I realized I also needed to add "is filled by" to the value list configuration key and I forgot to edit it. What I wanted to say is that it wasn't in the value list and that you might want to add it to other projects, since it's the only one missing. I've fixed it for MKAP.

**Comment by luizaogs** [comment]
I also just added two more things that came to mind today to the bullet point list above.

**Comment by urapadmin** [comment]
can we close this?

**Comment by luizaogs** [comment]
Well depends on whether you guys want to think about any of the suggestions/requests above or not, I guess. Only the “is filled by” issue has been addressed.

**Comment by urapadmin** [comment]
ah. I thought we went through it and either made single tickets or not. Then the answer is no. We have to work on it before we can close it.

**Comment by luizaogs** [comment]
We have this one: #2414 so I'll add it to the list above.

**Comment by lbestock** [comment]
I definitely think periods can be a fruitful place to think about how we use them and how to record them. That isn't captured well in the current recording and is fundamental to archaeological analysis. But it's also not usually a field thing to determine that. Issues of what is being dated - an object? an assemblage? a locus on the basis of multiple assemblages? a locus on the basis of stratigraphic relations? arise, and behind them the method of arriving at the date. So it even gets into specialist recording. 

I remain allergic to doing more with elevations because I don't think they are very useful at all in Kiosk. So before I want to think about whether or not to build in an elevation automatic trigger I would want to hear what MKAP has done with the numbers put into Kiosk. So it should be asked much more broadly, if and how elevations can be useful in this kind of recording. Tinkering with what wasn't great to begin with isn't very satisfying.

Basically and I am interested in this confluence: both elevations and period in Kiosk are there now as digital versions of what was done on paper better. Elevations because they can't just be a list, they need some X Y data on there to be useful. And period because it's interpretive and can be messy or neat depending on everything from how many objects we're talking about to the type of context to the type of analysis of the things that give the date. Currently it's attached just to CM and that of course is not the only way periodizing things works in archaeology.

**Comment by luizaogs** [comment]
As I mentioned above, I mostly use elevations to figure out profiles. So it can be a pain to flip through forms with empty fields because I didn't have time to write down the same numbers in both closings and openings while trying to supervise a bunch of specialists. But then, yes, it would be more of a convenience than necessarily useful for analysis later on, so I'm not sure it's worth it. I noted it because the way it is in the recording right now didn't work for my workflow and it annoyed me, but that doesn't necessarily mean you need to do anything about it.

**Comment by lbestock** [comment]
But let's think about that need from a different perspective. Why flip through forms at all? Wouldn't a report, or something like an HM with elevations on it, be more useful?

**Comment by luizaogs** [comment]
Probably! HM with elevations on it would be awesome.

**Comment by urapadmin** [comment]
this is one of those tickets that have kinda drowned in academic discussion. I am trying to summarize the things I can work with:

- #2414 

- a list of periods is a bit over the top here in my opinion. That is even a challenge to the current layout. Periods are rarely used at all by most projects. So I would rather go with extending the value list by the ranges you need. Object if this is super important to you.

- I don't have a good idea on recording opening elevations related to closing elevations automatically. How would I clearly determine what locus' closing elevation to use for an opening elevation of another locus? Even if I used the locus relations for that I am not sure this would be unequivocal in all situations. If I pick the wrong one as the initial value for opening elevations the danger is there that wrong values are recorded unnoticed. I can't come up with an idea here that'd make me happy. So automatic is out unless somebody has an idea how to safely determine the locus to use. 
- Now for the manual option. Hm. 
  - Is selecting a locus really easier than typing in the number into the field? 
  - Or is the issue that you would first have to look it up? 
  - Which loci would you select from? Are there criteria that would work? Like all directly related loci that are earlier or so? 
  - Would an option to show elevations in the list of locus relations help anyhow?
  - Would a simple qc rule (a hint or warning) help instead that reminds you that elevations are missing?
  - Instead of adding the elevations to the harris matrix (which I could do, of course, but I would not want to touch the hm module right now), I am wondering if a report that lists all loci of a unit with closing and opening elevations might already solve your need?

**Comment by luizaogs** [comment]
For the elevations, I don't know. It might be better to report back from the field. The reason why this came up at all was because both Didi and I left them out of the recording as a result of having to look them up, as I said above, which is not ideal. QC wouldn't help too much, I don't think - we knew they weren't there, it was just a matter of it being too difficult to flip between locus forms and copy decimals. Selecting a locus is certainly easier than typing in the decimals, and that would be my preferred manual solution, and the loci that would make sense for that would be anything above the new locus (if one locus is below the other, the closing of the above locus is the opening of the one below). But Laurel was skeptical above that this is even a real need so perhaps not worth your time.

**Comment by luizaogs** [comment]
For periods, that's fine. I think, as Laurel said above, that it's true that it's more for analysis anyway. 

What about the demarcation of uncertain dating? Not sure what you guys think about that.

And a field for comparanda was also something that was requested. Again perhaps more relevant for analysis, but since we currently do not have a CM analysis module on Kiosk, we are missing that in the field.

**Comment by urapadmin** [comment]
- I have no objections to marking uncertain dating (although I always wanted to make it a more general topic, but nevermind) so what would uncertain dating be? just a checkbox? 
- what would that field for comparanda contain? It means nothing to me. Can you elaborate on that? Is it really just a field?

**Comment by urapadmin** [comment]
are you using "full" elevations with single fields for NW, NE. SW, SE in MKAP or "simple" just with a big text block?

**Comment by lbestock** [comment]
A full-fledged comparanda tool is an analysis thing so not at all sure my thoughts even belong in this discussion for MKAP for right now. It's long term. But I would want:

- a separate N images (that's not hypothetical - I once found an object I didn't want to lose track of for comparison, and put the image in URAP in the only place I had, in the CM I wanted to compare it to. Bad. Confused me immediately.)
- bibliography
- notes
- urls with links
- museum and museum number so I can sort by museum and give myself a list, should I be in that museum, of things to look at

And that's on the CM front. Another reason this is an analysis thing, not a recording thing, is that all sorts of things can have comparanda. As in, from site to unit to context to CM.

**Comment by urapadmin** [comment]
okay, you may make a new ticket for that. That is clearly nothing that can or should be built into the recording app right now.
The question here is:
- is there anything feasible that would actually help MKAP users in their current life? Or is it wasted time until it can be tackled bigger as described by Laurel?

**Comment by luizaogs** [comment]
> * I have no objections to marking uncertain dating (although I always wanted to make it a more general topic, but nevermind) so what would uncertain dating be? just a checkbox?

Yes, I think a checkbox for uncertain dating works. That's what the registrar asked for.

**Comment by luizaogs** [comment]
> okay, you may make a new ticket for that. That is clearly nothing that can or should be built into the recording app right now. The question here is:
> 
> * is there anything feasible that would actually help MKAP users in their current life? Or is it wasted time until it can be tackled bigger as described by Laurel?

Yes. A field for "comparanda" in the CM recording would help. Last year we wanted to note that we found a bowl with decoration similar to some found in other sites and we didn't have anywhere to put it. Ideally there would be separate places for comparanda from the same site and comparanda from other sites, but I think a single field would suffice for now, since clearly there is more to think about on the comparanda front. And then we could have a query that collects objects with comparanda. 

This was something we wanted to do in the field, by the way. During registration. Not during analysis. So while I agree that comparanda is much bigger than field recording - it is also useful in the field.

**Comment by luizaogs** [comment]
> are you using "full" elevations with single fields for NW, NE. SW, SE in MKAP or "simple" just with a big text block?

Full.

**Comment by urapadmin** [comment]
> This was something we wanted to do in the field, by the way. During registration. Not during analysis. So while I agree that comparanda is much bigger than field recording - it is also useful in the field.

just a thought: what about a field "interpretation" or "research notes" in collected material similar to what we have in locus? Such a field would get everything for now that is no supposed to go into the cm description. So e.g. information for comparanda. But it widens the use case a bit for other information. 

Is this for small finds only or can it occur for all collected material types (preferable for me from a technical point of view)?

**Comment by luizaogs** [comment]
"Research notes" would be preferable. Small finds would probably be most common, but all CM types I think makes sense.

**Comment by urapadmin** [comment]
### recording app 16.2 / Kiosk 1.6.5.2 (on test) or Kiosk 1.6.2.3 (other projects).
- collected material has a tab in a tab now showing description  and registration in separate tabs and "research notes" in a third tab
- research notes are only accessible if visibility/show_research_notes is set to yup. I can see that some projects don't want this (arch1900 e.g.). By default the field is accessible.
- research notes show a help button if help/help_research_notes is set to a help text. That help text would be shown.
- research notes have a reference navigation button and can be enlarged
- period has an "uncertain dating" checkbox
- storage now has a reference navigation button for what its worth

As these are new fields and a whole new table please check synchronization.

This is only for the recording app, some of these changes need to be reflected in Q&V: #2891

**Comment by luizaogs** [comment]
Like in the to-do list, the reference navigation buttons are there but are not actually working yet: 

![IMG_0468](https://github.com/user-attachments/assets/08a82cef-f2f9-4f4f-ae17-c6ca9f96c5aa)

![IMG_0469](https://github.com/user-attachments/assets/193f4e89-1332-451c-a021-f95f50160faa)

**Comment by luizaogs** [comment]
As you can see in the screenshot above, the "uncertain dating" is also not there yet (unless I'm not finding it).

**Comment by luizaogs** [comment]
Navigation also not yet working in the storage field.

**Comment by luizaogs** [comment]
Synchronization also didn't work. The research notes I added about cheese above are gone after I downloaded the workstation again. (What I added in storage is there though, so I know this is the right database)

**Comment by urapadmin** [comment]
> As you can see in the screenshot above, the "uncertain dating" is also not there yet (unless I'm not finding it).

it appears once you actually have a period

**Comment by luizaogs** [comment]
It does. Sorry. Ignore that.

**Comment by urapadmin** [comment]
> Navigation also not yet working in the storage field.

identifiers need quotes o[ brackets to be found. If they contain a dash it also works, but the rule of thumb is to mark them

**Comment by luizaogs** [comment]
Fine. Just synchronization, then.

**Comment by urapadmin** [comment]
the date field isn't okay. It should accept latin inputs and show the latin representation below.

**Comment by urapadmin** [comment]
### Kiosk 1.6.5.2
- no new Kiosk necessary. It was just a typo in a configuration file. Now both release notes and uncertain dating should sync

**Comment by urapadmin** [comment]
> For the elevations, I don't know. It might be better to report back from the field. The reason why this came up at all was because both Didi and I left them out of the recording as a result of having to look them up, as I said above, which is not ideal. QC wouldn't help too much, I don't think - we knew they weren't there, it was just a matter of it being too difficult to flip between locus forms and copy decimals. Selecting a locus is certainly easier than typing in the decimals, and that would be my preferred manual solution, and the loci that would make sense for that would be anything above the new locus (if one locus is below the other, the closing of the above locus is the opening of the one below). But Laurel was skeptical above that this is even a real need so perhaps not worth your time.

- For Luiza: see #2900
